packages/crawler/src/evaluate-page.ts

```diff
@@ -0,0 +1,155 @@
+      const tagNames = Object.keys(pdsTagNamesWithPropertyNames);
+
+      const getPdsComponentNameByPrefix = (el: Element, prefix: string): TagName | null => {
+        const tagName = el.tagName.toLowerCase();
+        return (
+          (Object.keys(pdsTagNamesWithPropertyNames).find(
+            (compName) => (prefix ? `${prefix}-${compName}` : compName) === tagName
+          ) as TagName) || null
+        );
+      };
+
+      const getAllChildElements = (el: Element): Element[] => {
+        const children = Array.from(el.children)
+          .concat(Array.from(el.shadowRoot?.children || []))
+          .flat() as Element[];
+        return children.concat(children.map(getAllChildElements).flat());
+      };
+
+      const getChildren = (el: Element): string | null =>
+        Array.from(el.children).reduce((result, slotEl) => {
+          if (slotEl.children.length) {
+            const copy = slotEl.cloneNode(true) as Element;
+            // we save only the highest dom level in the reports, in order not to make them too big
+            copy.innerHTML = '(the content was stripped)';
+            return result + copy.outerHTML;
+          }
+          return result + slotEl.outerHTML;
+        }, '' as string) || el.textContent;
+
+      const getHostPdsComponent = (el: Element, prefix: string): TagName | null => {
+        const rootNode = (el.getRootNode() as ShadowRoot).host;
+        if (rootNode) {
+          return getPdsComponentNameByPrefix(rootNode, prefix);
+        }
+        return null;
+      };
+
+      const parseIfJSON = (jsonStr: string | number | symbol): object | string | number | symbol => {
+        if (typeof jsonStr === 'string' && jsonStr.includes('{')) {
+          try {
+            // jsonStr is potentially JSON parsable string, e.g. "{ base: 'block', l: 'inline' }"
+            return JSON.parse(
+              jsonStr
+                .replace(/'/g, '"') // convert single quotes to double quotes
+                .replace(/[\s"]?([a-z\-]+)[\s"]?:([^//])/g, '"$1":$2') // wrap keys in double quotes if they don't have them but ignore potential urls
+            );
+          } catch {
+            // jsonStr is string, e.g. "block" or "inline"
+            return jsonStr;
+          }
+        } else {
+          // jsonStr is object, e.g. { base: 'block', l: 'inline' } or number, e.g. 123 or boolean, e.g. true
+          return jsonStr;
+        }
+      };
+
+      const getAllConsumedProperties = <
+        PComponentName extends keyof HTMLElementTagNameMap,
+        PComponentElement extends HTMLElementTagNameMap[PComponentName],
+        PComponentPropertyName extends keyof PComponentElement,
+        PComponentPropertyValue extends keyof PComponentElement[PComponentPropertyName]
+      >(
+        el: Element,
+        allPdsPropertiesForComponentName: string[]
+      ) => {
+        const pEl = el as PComponentElement;
```

Comment:
Alias might lead to confusion.

Reply:
true.. I was also thinking that it might lead to confusion.. But this alias is actually logical, because we take an Element and consider it to be "PDS Component" with corresponding properties.. Added comment in code to make it more clear:
`// we consider this element as PDS Component, therefore we need this alias here`